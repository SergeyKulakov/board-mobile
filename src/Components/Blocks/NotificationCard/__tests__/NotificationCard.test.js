import React from 'react'
import { shallow } from 'enzyme'

import 'jest-styled-components'

import Component from '../NotificationCard'

describe('NotificationCard', () => {
  const defaultProps = {
    type: 'type',
    date: 1571651436414,
    user: {
      idVerified: '1234sdqwe',
      username: 'userId',
      avatarURL: 'avatarUrl',
    },
    job: {
      _id: 'jobId',
      pics: ['pick1', 'pick2'],
    },
    reason: 'reason',
    isRead: false,
    t: text => text,
    onClickUser: () => null,
    onClickJob: () => null,
  }

  it('render snapshot', () => {
    const wrapper = shallow(<Component {...defaultProps} />)
    const withoutJob = shallow(<Component {...defaultProps} job={null} />)
    const withoutData = shallow(
      <Component {...defaultProps} job={null} user={null} />,
    )

    expect([wrapper, withoutJob, withoutData]).toMatchSnapshot()
  })
})
