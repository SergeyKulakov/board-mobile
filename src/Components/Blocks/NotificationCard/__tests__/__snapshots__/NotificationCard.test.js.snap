// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCard render snapshot 1`] = `
Array [
  <Styled(View)>
    <Styled(View)>
      <Styled(View)>
        <Styled(View)>
          <Image
            data="https://s3.amazonaws.com/spotjobs-images-097579889258-us-east-1/job-pics/jobId/pick1"
            resizeMode="center"
            withLoading={true}
          />
        </Styled(View)>
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <Component
        job={
          Object {
            "_id": "jobId",
            "pics": Array [
              "pick1",
              "pick2",
            ],
          }
        }
        onClickJob={[Function]}
        onClickUser={[Function]}
        reason="reason"
        type="type"
        user={
          Object {
            "avatarURL": "avatarUrl",
            "idVerified": "1234sdqwe",
            "username": "userId",
          }
        }
      />
      <Styled(View)>
        <Styled(Text)>
          notifications.on
        </Styled(Text)>
        <Styled(Text)
          isBold={true}
        >
          21/10/2019
        </Styled(Text)>
        <Styled(Text)>
          notifications.at
        </Styled(Text)>
        <Styled(Text)
          isBold={true}
        >
          12:50 PM
        </Styled(Text)>
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <Styled(Icon) />
    </Styled(View)>
  </Styled(View)>,
  <Styled(View)>
    <Styled(View)>
      <Styled(View)>
        <ProviderPhoto
          avatarURL="avatarUrl"
          isCheck="1234sdqwe"
          username="userId"
        />
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <Component
        job={null}
        onClickJob={[Function]}
        onClickUser={[Function]}
        reason="reason"
        type="type"
        user={
          Object {
            "avatarURL": "avatarUrl",
            "idVerified": "1234sdqwe",
            "username": "userId",
          }
        }
      />
      <Styled(View)>
        <Styled(Text)>
          notifications.on
        </Styled(Text)>
        <Styled(Text)
          isBold={true}
        >
          21/10/2019
        </Styled(Text)>
        <Styled(Text)>
          notifications.at
        </Styled(Text)>
        <Styled(Text)
          isBold={true}
        >
          12:50 PM
        </Styled(Text)>
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <Styled(Icon) />
    </Styled(View)>
  </Styled(View)>,
  <Styled(View)>
    <Styled(View)>
      <Styled(View)>
        <Component />
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <Component
        job={null}
        onClickJob={[Function]}
        onClickUser={[Function]}
        reason="reason"
        type="type"
        user={null}
      />
      <Styled(View)>
        <Styled(Text)>
          notifications.on
        </Styled(Text)>
        <Styled(Text)
          isBold={true}
        >
          21/10/2019
        </Styled(Text)>
        <Styled(Text)>
          notifications.at
        </Styled(Text)>
        <Styled(Text)
          isBold={true}
        >
          12:50 PM
        </Styled(Text)>
      </Styled(View)>
    </Styled(View)>
    <Styled(View)>
      <Styled(Icon) />
    </Styled(View)>
  </Styled(View)>,
]
`;
